diff --git a/CMakeLists.txt b/CMakeLists.txt
index a16748630..31464ab2e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,11 +67,6 @@ option(USE_UBSAN "Build with UndefinedBehaviorSanitizer" OFF)
 option(ENABLE_COVERAGE "Build with code coverage" OFF)
 option(UBSAN_HALT_ON_ERROR "Halt on error when building with UBSan" ON)
 
-if(${CMAKE_VERSION} VERSION_LESS "3.1.0" AND BUILD_SHARED_LIBS)
-	# scap_engine_savefile uses target_sources
-	message(FATAL_ERROR "Shared libraries requires CMake 3.1 or later.")
-endif()
-
 include(GNUInstallDirs)
 
 # Add path for custom CMake modules.
@@ -150,3 +145,49 @@ if(CREATE_TEST_TARGETS)
 	endif()
 
 endif()
+
+# --------------------------------
+# Install the libraries
+install(TARGETS cri_v1
+                cri_v1alpha2
+                driver_event_schema
+                pman
+                scap
+                sinsp
+                scap_engine_bpf
+                scap_engine_gvisor
+                scap_engine_kmod
+                scap_engine_modern_bpf
+                scap_engine_nodriver
+                scap_engine_noop
+                scap_engine_savefile
+                scap_engine_source_plugin
+                scap_engine_test_input
+                scap_engine_util
+                scap_error
+                scap_event_schema
+                scap_platform
+                scap_platform_util
+        EXPORT falcosecurity-libs-targets
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDE}
+)
+
+# Generate and install the necessary CMake files
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    falcosecurity-libsConfigVersion.cmake
+    VERSION 0.15.0
+    COMPATIBILITY AnyNewerVersion
+)
+
+install(EXPORT falcosecurity-libs-targets
+        FILE falcosecurity-libsTargets.cmake
+        NAMESPACE falcosecurity::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/falcosecurity-libs)
+
+install(FILES falcosecurity-libsConfig.cmake
+              ${CMAKE_CURRENT_BINARY_DIR}/falcosecurity-libsConfigVersion.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/falcosecurity-libs)
